File: Algorithms/p334_Increasing_Triplet_Subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int v1 = std::numeric_limits<int>::max();
        int v2 = std::numeric_limits<int>::max();
        for(int v : nums){
            if(v<=v1)      v1 = v;
            else if(v<=v2) v2 = v;
            else           return 1;
        }
        return 0;
    }
};

File: Algorithms/p334_Increasing_Triplet_Subsequence/p334_Increasing_Triplet_Subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        v1 = float('inf')
        v2 = float('inf')
        for v in nums:
            if v<=v1:
                v1 = v
            elif v<=v2:
                v2 = v
            else:
                return True
        return False
